src/logging.cpp

```diff
@@ -190,29 +197,37 @@ std::string BCLog::Logger::LogTimestampStr(const std::string &str)
     } else
         strStamped = str;
 
-    if (!str.empty() && str[str.size()-1] == '\n')
-        m_started_new_line = true;
-    else
-        m_started_new_line = false;
-
     return strStamped;
 }
 
+std::string BCLog::Logger::LogThreadnameStr(const std::string &str)
+{
+    if (!m_log_threadnames || !m_started_new_line) {
+        return str;
+    }
+    std::string thread_name = thread_util::GetInternalName();
+    // The longest thread name (with numeric suffix) we have at the moment is 13 characters.
+    thread_name.resize(13, ' ');
+    return "[" + std::move(thread_name) + "] " + str;
+}
+
 void BCLog::Logger::LogPrintStr(const std::string &str)
 {
-    std::string strTimestamped = LogTimestampStr(str);
+    std::string strPrefixed = LogThreadnameStr(LogTimestampStr(str));
```

Comment:
It seems more natural to me to have the thread name after the timestamp.

This way round breaks combine_logs without a fix. I imagine other log parsers expect the timestamp at the start of the line as well.

Reply:
Fixed